### DISCUSSION TITLE
Does CVE-2022-42889 (commons-text) affect dropwizard?

### DISCUSSION THREAD

**Discussion by natan-abolafya** [dropwizard/dropwizard#6094] [Q&A] [answered]
This CVE is released recently https://nvd.nist.gov/vuln/detail/CVE-2022-42889 and it seems to be quite severe.

I am investigating if our product is affected by this vulnerability so we know whether to patch the older versions of our software and release security advisory. Our older versions have transitive dependency to apache commons-text library via dropwizard. So I'm wondering, if the way dropwizard uses commons-text is affected by it.

Note: It seems dropwizard 2.0.33 still uses a vulnerable version. 2.1 is fine but for those who prefers to stay at 2.0, it might be good to release a new patch.

**Comment by joschi** [comment]
Hi Natan, thanks for bringing this up! 👍 

The latest release of Dropwizard 2.0.x indeed is still using a vulnerable version of Apache Commons Text.

The version has already been bumped by @renovate-bot in https://github.com/dropwizard/dropwizard/pull/5951, but we didn't publish a new release yet.

We'll do so very soon to make sure that developers don't accidentally use a vulnerable version.

**Reply by natan-abolafya** [reply]
Hi @joschi, thanks for the update.

Do you know if dropwizard uses the library in a way that can be exploited? For example if it's used in the configuration parsing, that could be an issue though pretty mild since having access to the configuration file is pretty hard from the attacker point of view.


I have a feeling this CVE will be the next log4j, so it's good to be on top of it.

**Reply by joschi** [reply]
@natan-abolafya As far as I see, only [`EnvironmentVariableSubstitutor`](https://github.com/dropwizard/dropwizard/blob/v2.0.33/dropwizard-configuration/src/main/java/io/dropwizard/configuration/EnvironmentVariableSubstitutor.java) would be affected by this vulnerability and this class has to be explicitly used in a Dropwizard application:

https://www.dropwizard.io/en/release-2.0.x/manual/core.html#environment-variables

**Reply by joschi** [reply]
After more consideration, I'd say Dropwizard is not affected at all because it wasn't using the string interpolation functionality, only string substitution. 🤔

**Reply by natan-abolafya** [reply]
This is great! Thank you.

**Comment by joschi** [comment, marked as answer]
Dropwizard 2.0.34 has been released which comes with Apache Commons Text 1.10.0.

https://github.com/dropwizard/dropwizard/releases/tag/v2.0.34
